app/controllers/ListEventsController.js

```diff
@@ -9,10 +10,21 @@ GameUpApp.controller('ListEventsController', ['$scope', '$location', function($s
     return query.find();
   };
 
+  $scope.retrievePromotedEventList = function(){
```

Comment:
+1 @ajitpawar 

Also, I would consider defining the `retrievePromotedEventList` as follows:

```js
$scope.retrievePromotedEventList = function(options){
    // ...
```

Where `options` is an object that specifies the query parameters.      
For example:

```js
options = {
    limit: 5,
    location: {lat: 40.0, lon: -30.0}
} 
```